large-message-core/src/main/java/com/bakdata/kafka/CompressionType.java

```diff
@@ -0,0 +1,142 @@
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.SNAPPY, bytes);
+        }
+    },
+
+    LZ4(3, "lz4") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+    },
+
+    ZSTD(4, "zstd") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+    };
+
+    private static byte[] compress(org.apache.kafka.common.record.CompressionType compressionType, byte[] bytes) {
+        ByteBufferOutputStream outStream = new ByteBufferOutputStream(bytes.length);
+        try (OutputStream stream = compressionType.wrapForOutput(outStream, RecordBatch.MAGIC_VALUE_V2)) {
+            stream.write(bytes);
+            stream.flush();
+        } catch (IOException e) {
+            throw new SerializationException("Failed to compress with type " + compressionType, e);
+        }
+
+        return outStream.buffer().array();
+    }
+
+    private static final BufferSupplier bufferSupplier = BufferSupplier.create();
+
+    private static byte[] decompress(org.apache.kafka.common.record.CompressionType compressionType, byte[] bytes) {
+        try (InputStream stream = compressionType.wrapForInput(ByteBuffer.wrap(bytes), RecordBatch.MAGIC_VALUE_V2, bufferSupplier)) {
+            return IoUtils.toByteArray(stream);
+        } catch (IOException e) {
+            throw new SerializationException("Failed to compress with type " + compressionType, e);
+        }
+    }
+
+    public final byte id;
+    public final String name;
+
+    CompressionType(int id, String name) {
+        this.id = (byte)id;
+        this.name = name;
+    }
+
+    public static final String HEADER_NAME = HeaderLargeMessagePayloadProtocol.HEADER_PREFIX + ".compression";
+
+    public static CompressionType forId(int id) {
+        switch (id) {
+            case 0:
+                return NONE;
+            case 1:
+                return GZIP;
+            case 2:
+                return SNAPPY;
+            case 3:
+                return LZ4;
+            case 4:
+                return ZSTD;
+            default:
+                throw new IllegalArgumentException("Unknown compression type id: " + id);
+        }
+    }
+
+    public static CompressionType forName(String name) {
+        if (NONE.name.equals(name))
```

Comment:
You could also use a switch statement

Reply:
switch statement doesn't work here, because the compiler doesn't know that `NONE.name` is a constant (despite `name` being final)